### PR TITLE
fix: Standardize headings, add spacing and add button to view more speakers 

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/6737
Screenshot from 2021-02-17 18-32-54
Screenshot from 2021-02-17 18-32-50
Screenshot from 2021-02-17 18-32-45
Screenshot from 2021-02-17 18-25-51
Screenshot from 2021-02-17 18-25-43
Screenshot from 2021-02-17 18-22-46
Screenshot from 2021-02-17 18-22-36
Screenshot from 2021-02-17 18-22-06

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
Thanks!

1. There is no "More Speakers" button on the event I tested here yet https://open-event-frontend-git-fork-maze-runnar-timezone.eventyay.vercel.app/e/fa96ae2c
2. Please also add a slight bit more spacing above the heading 2 and ensure all heading 2 have the same spacing, e.g. the heading "Online Event" does not seem to have spacing above yet.
Screenshot from 2021-02-17 14-17-16

**Comment by maze-runnar** [comment]
> There is no "More Speakers" button on the event I tested here yet https://open-event-frontend-git-fork-maze-runnar-timezone.eventyay.vercel.app/e/fa96ae2c

the server change is also pending for this. once that will merge it will shoow correctly.https://github.com/fossasia/open-event-server/pull/7752

**Comment by iamareebjamal** [comment]
Please use count from API response. The server PR is not needed and does not contain correct logic for filtering deleted and accepted/confirmed speakers

**Comment by maze-runnar** [comment]
> Please also add a slight bit more spacing above the heading 2 and ensure all heading 2 have the same spacing, e.g. the heading "Online Event" does not seem to have spacing above yet.

I have added `mt-8(margin-top: 2rem(32px))` everywhere for ``h2`` here to `` <h2 class="mt-8">{{t 'Online Event'}}</h2> ``

**Comment by mariobehling** [comment]
I am updating the branch to see this change with all recent changes.

**Comment by mariobehling** [comment]
I am confused because the spacing looks different everywhere, probably because there is a spacing on the bottom of some areas as well. I wonder how this can be standardized.

The spacing above "Featured Speakers" and above "Online Event" looks definitely different to the spacing above "Supported By", "Organized By" and "Getting Here". What can be done to make this same across the page?

How can the spacing above Featured Speakers and Online Event be the same as the other areas?

**Comment by mariobehling** [comment]
But: Seeing the change in the table "Tickets", it is also clear that it does not make sense to have a heading spacing inside a table. Therefore the heading 2 in the table should not have the spacing. Could you make that change, please?

Screenshot from 2021-02-17 23-51-04

**Comment by maze-runnar** [comment]
> Please use count from API response. 

I am not Not understanding this.

**Comment by iamareebjamal** [comment]
Featured Speaker Query:
image

Has Speakers Query:
image

Can you see the count in the screenshot?

**Comment by maze-runnar** [comment]
yes :no_mouth:

**Comment by maze-runnar** [comment]
using count is causing some problems, will discuss in meeting.

**Comment by mariobehling** [comment]
Nice. More speakers is working now.

But the spacing above heading 2 does not look sufficient. Could you double the space, please?

**Comment by maze-runnar** [comment]
> Could you double the space, please?

ok

**Comment by maze-runnar** [comment]
Screenshot from 2021-02-18 21-43-02
Screenshot from 2021-02-18 21-41-56
Screenshot from 2021-02-18 21-41-50
Screenshot from 2021-02-18 21-41-45

**Comment by mariobehling** [comment]
Thanks it is ok for me for now, but we need to come back to this later. Spacing between we sections is different also on mobile page. It is now better than before.

File: app/components/public/featured-speaker-list.js [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+export default class FeaturedSpeakerList extends Component {
+
+    @tracked
+    allSpeakers = null;
+    async didInsertElement() {
+        super.didInsertElement(...arguments);
+        this.countSpeakers();
+    }
+    async countSpeakers() {
+       this.allSpeakers = (await this.loader.load(`/events/${this.event.id}/speakers?cache=true&public=true&fields[speaker]=id&page[size]=1&filter=${JSON.stringify(SPEAKERS_FILTER)}`)).meta.count;
```
Comment (iamareebjamal) [inline app/components/public/featured-speaker-list.js]: No need to add one more query.

File: app/components/public/featured-speaker-list.js [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+    @tracked
+    allSpeakers = null;
+
+    async didInsertElement() {
+      super.didInsertElement(...arguments);
+      this.countSpeakers();
+    }
+
+    async countSpeakers() {
+      this.allSpeakers = (await this.loader.load(`/events/${this.event.id}/speakers?cache=true&public=true&fields[speaker]=id&page[size]=1&filter=${JSON.stringify(SPEAKERS_FILTER)}`)).meta.count;
```
Comment (iamareebjamal) [inline app/components/public/featured-speaker-list.js:18]: No need to add one more query

**Comment by mariobehling** [comment]
@maze-runnar Let's finalize this PR, please.

File: app/components/public/featured-speaker-list.js [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+
+    @tracked allSpeakers = null;
+
+    async didInsertElement() {
+      super.didInsertElement(...arguments);
+      this.countSpeakers();
+    }
+
+    async countSpeakers() {
+      this.allSpeakers = this.allSpeakers ?? await hasSpeakers(this.loader, this.event, true);
```
Comment (iamareebjamal) [inline app/components/public/featured-speaker-list.js:17]: No need to add one more query as discussed in meeting several times

File: app/utils/event.ts [hunk 4]
```diff
@@ -2,8 +2,12 @@ import Event from 'open-event-frontend/models/event';
 import { SPEAKERS_FILTER } from 'open-event-frontend/routes/public/speakers';
 import Loader from 'open-event-frontend/services/loader';
 
-export async function hasSpeakers(loader: Loader, event: Event): Promise<number> {
-  return (await loader.load(`/events/${event.id}/speakers?cache=true&public=true&fields[speaker]=id&page[size]=1&filter=${JSON.stringify(SPEAKERS_FILTER)}`)).data.length;
+export async function hasSpeakers(loader: Loader, event: Event, needCount: boolean): Promise<number> {
+  if (needCount) {
+    return (await loader.load(`/events/${event.id}/speakers?cache=true&public=true&fields[speaker]=id&page[size]=1&filter=${JSON.stringify(SPEAKERS_FILTER)}`)).meta.count;
+  } else {
+    return (await loader.load(`/events/${event.id}/speakers?cache=true&public=true&fields[speaker]=id&page[size]=1&filter=${JSON.stringify(SPEAKERS_FILTER)}`)).data.length;
```
Comment (iamareebjamal) [inline app/utils/event.ts:9]: No need for duplication

File: app/components/public/featured-speaker-list.js (hunk 3 above)
Comment (maze-runnar) [reply app/components/public/featured-speaker-list.js:17]: it's not working that way, I have tried that.

File: app/utils/event.ts (hunk 4 above)
Comment (maze-runnar) [reply app/utils/event.ts:9]: returned `return (await loader.load(`/events/${event.id}/speakers?cache=true&public=true&fields[speaker]=id&page[size]=1&filter=${JSON.stringify(SPEAKERS_FILTER)}`));` here and used `hasSpeaker.meta.count` but it doesn't work.

File: app/components/public/featured-speaker-list.js (hunk 3 above)
Comment (maze-runnar) [reply app/components/public/featured-speaker-list.js:17]: But we have to use `hasSpeaker` somewhere ? without adding it no where how it will get value.

Comment (iamareebjamal) [reply app/components/public/featured-speaker-list.js:17]: It is already called in side menu, move it to public controller

**Comment by mariobehling** [comment]
Merging dev branch to see how this looks like with the recent changes. Hope it can be finished now.

**Comment by iamareebjamal** [comment]
@maze-runnar Please see the solution

**Review by mariobehling — APPROVED** [review]

**Pull request merged**